x-pack/plugins/security_solution/public/common/components/alert_count_by_status/alert_count_by_rule_by_status.tsx

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo } from 'react';
+
+import type { EuiBasicTableColumn } from '@elastic/eui';
+import { EuiBasicTable, EuiEmptyPrompt, EuiLink, EuiPanel, EuiToolTip } from '@elastic/eui';
+import { euiStyled } from '@kbn/kibana-react-plugin/common';
+
+import type { Status } from '../../../../common/detection_engine/schemas/common';
+import { SecurityPageName } from '../../../../common/constants';
+import { useNavigateToTimeline } from '../../../overview/components/detection_response/hooks/use_navigate_to_timeline';
+import { useQueryToggle } from '../../containers/query_toggle';
+import type { NavigateTo, GetAppUrl } from '../../lib/kibana';
+import { useNavigation } from '../../lib/kibana';
+import { FormattedCount } from '../formatted_number';
+import { HeaderSection } from '../header_section';
+import { HoverVisibilityContainer } from '../hover_visibility_container';
+import { BUTTON_CLASS as INSPECT_BUTTON_CLASS } from '../inspect';
+import { LastUpdatedAt } from '../last_updated_at';
+import { useLocalStorage } from '../local_storage';
+import { MultiSelectPopover } from './components';
+import * as i18n from './translations';
+import type { AlertCountByRuleByStatusItem } from './use_alert_count_by_rule_by_status';
+import { useAlertCountByRuleByStatus } from './use_alert_count_by_rule_by_status';
+
+interface AlertCountByStatusProps {
+  field: string;
+  value: string;
+  signalIndexName: string | null;
+}
+
+interface StatusSelection {
+  [fieldName: string]: Status[];
+}
+
+type GetTableColumns = ({
+  getAppUrl,
+  navigateTo,
+  openRuleInTimelineWithAdditionalFields,
+}: {
+  getAppUrl: GetAppUrl;
+  navigateTo: NavigateTo;
+  openRuleInTimelineWithAdditionalFields: (ruleName: string) => void;
+}) => Array<EuiBasicTableColumn<AlertCountByRuleByStatusItem>>;
+
+const KIBANA_RULE_ALERT_FIELD = 'kibana.alert.rule.name';
+const statuses = ['open', 'acknowledged', 'closed'];
```

Comment:
for coherence, we should name it `STATUSES`. Also, it would be nice to make it read-only by adding `as const` at the end.

Reply:
there is already a type `Status` in `x-pack/plugins/security_solution/common/detection_engine/schemas/common/schemas.ts`

Reply:
changed!